datafusion/src/physical_plan/sort_preserving_merge.rs

```diff
@@ -246,7 +274,19 @@ impl SortKeyCursor {
             .zip(other.columns.iter())
             .zip(options.iter());
 
-        for ((l, r), sort_options) in zipped {
+        // Recall or initialise a collection of comparators for comparing
+        // columnar arrays of this cursor and "other".
+        let cmp = self
+            .batch_comparators
+            .entry(other.batch_idx)
+            .or_insert_with(|| Vec::with_capacity(other.columns.len()));
```

Comment:
I was initially worried that mutating `cmp` below might mask errors (e.g. that the number of columns had grown somehow), so I wonder if it is worth putting the initialization of `cmp` into this callsite (rather than the loop below)?

However, perhaps that avoids having to clone or collect `zipped`.

Please feel free to ignore this comment

Reply:
Ha. Yeah that's how I had it but you have to re-create the iterator. I'm happy to go with whatever

Reply:
I think this way is fine, personally